### ISSUE TITLE
V compiler failing to work properly with cyrillic paths.

### ISSUE THREAD

**Issue by KatPurpy** [vlang/v#8683]
v doctor output
```
OS: windows, Microsoft Windows 8.1 Pro v9600 64-ࠧ�來��
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-6300 CPU @ 3.80GHz
CC version: cc (Rev1, Built by MSYS2 project) 10.2.0

getwd: C:\CompilationCrap\v
vmodules: C:\Users\[REDACTED]\.vmodules
vroot: C:\CompilationCrap\v
vexe: C:\CompilationCrap\v\v.exe
vexe mtime: 2021-02-08 18:39:43
is vroot writable: true
is vmodules writable: false
V full version: V 0.2.2 09cff69

Git version: git version 2.19.1.windows.1
Git vroot status: Error: fatal: not a git repository (or any of the parent directories): .git
.git/config present: false
thirdparty/tcc status: thirdparty-windows-amd64 f7691963
```

**What did you do?**
1. Tried to use ```v help``` while the executable is under folder with cyrilic characters.
2. Tried to compile tetris example after moving the executable to C:/CompilationCrap/v folder. The command was ```v run examples/tetris/tetris.v```

**What did you expect to see?**
1. Expected to see help
2. Expected to see no errors and have tetris example running
 
**What did you see instead?**
1.
C:\Users\[REDACTED]\Downloads\v>v help
V Error: Unknown help topic provided. Use `v help` for usage information.
2. 
gcc: error: C:\Users\[REDACTED]\.vmodules\cache\a0\a084424f774d8f05340437d1c93d0b91.o: No such file or directory
gcc: error: C:\Users\[REDACTED]\.vmodules\cache\4a\4a44544dbe66bd1cf288cf0b9ef7e0d1.o: No such file or directory
[a lot of = that were removed because markdown engine parses them inadequately]
builder error:

C error. This should never happen.

The [REDACTED] folder is a name in cyrillic censored for privacy reasons

The version is the latest one, precompiled, downloaded from the official website

**Comment by cbracketdash** [comment]
Try doing a clean install. Tell us if you still have the same error.

**Comment by KatPurpy** [comment]
What does clean install mean? Cloning from git repo and compiling manually?

**Comment by KatPurpy** [comment]
@cbracketdash Nope, same error.

**Comment by cbracketdash** [comment]
@KatPurpy Yes, deleted every download, path, temp file related to V and reinstalling it.

**Comment by KatPurpy** [comment]
not sure if I did clean it up properly i just copied V from official website but the problem is clearly presence of cyrillic characters

**Comment by KatPurpy** [comment]
by the way i just checked that in fact object files do exist in the directory, it is now clear that the trouble is somewhere with encoding and passing proper argument to gcc

**Comment by JalonSolov** [comment]
Can you try with `-cc tcc` and see if it has the same problem?

Also, try gcc from https://winlibs.com instead of the msys2 one you have.

**Comment by KatPurpy** [comment]
@JalonSolov I really doubt that would help since building tetris example fails as well as anything that requires linking to third party libraries. The other problem I found out that if the current working dir of the compiler contains cyrillic characters it can't do anything
![изображение](https://user-images.githubusercontent.com/43522532/107812577-f2b32400-6d80-11eb-99f6-4c6697bff7ba.png)

**Comment by cbracketdash** [comment]
Maybe just don't use Cyrillic characters?

**Comment by Delta456** [comment]
> 
> 
> Maybe just don't use Cyrillic characters?

IMO it should work with Cyrillic characters.

**Comment by AestheticalZ** [comment]
Tried on Linux; issue doesnt happen. Seems to be a Windows only issue.

**Comment by medvednikov** [comment]
Still an issue on Windows.

**Comment by JalonSolov** [comment]
I wonder if this is still a problem.  I don't have Windows, so can't test it.

I do know tcc used to have problems with any paths that had non-ASCII characters.  I thought this was fixed in later versions.

**Comment by revosw** [comment]
tcc uses `fopen` to open files given a path. The windows build of tcc has been built with msvcrt.dll in mind, so fopen calls into msvcrt.dll. Inside msvcrt.dll, you got two variants, the standard `fopen` and the non-standard but unicode-aware `_wfopen`. fopen is only made with ANSI in mind, so behind the scenes CreateFileA() is called.

When calling an ANSI function, the string input is interpreted based on the code page of the system locale you have configured Windows to be in[^1]. So for example if the system locale is LCID 1041 (japanese, japan) and I run an executable that calls CreateFileA(), the `char*` parameters will be interpreted using the CP932 ANSI code page. If the system locale is a western one such as LCID 1033 (english, united states), LCID 1036 (french, france) or similar, the CP1252 ANSI code page will be used.

CP1252 does not define cyrillic characters in its character set. CP1251 is the ANSI code page for cyrillic characters. Since there are no cyrillic characters mapped in CP1252, each missing character is replaced by `?`

```
v run .\8683-v-compiler-Проблема\
================== C compilation error (from tcc): ==============
cc: tcc: error: listfile 'C:\Users\simen\AppData\Local\Temp\v_0\8683-v-compiler-????????.01KHF99JJRA6R3X60EPPWEWX77.tmp.c.rsp' not found
=================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

There is a beta feature[^2] that, when enabled, makes all `___A()` windows functions interpret `char*` as UTF-8 encoded. It works by overriding the locale code page with the CP65001 code page, which is the code page for UTF-8. This lets me actually call fopen with cyrillic letters, even when my configured system locale uses CP1252

<img width="655" height="264" alt="Image" src="https://github.com/user-attachments/assets/4d3fd75d-3e04-4042-b6cc-b0a5f8a23126" />

There's absolutely nothing V can do about this. The solution for your case is to toggle on the beta feature. Note that enabling it won't magically transcode input text (for example from a file) that has been encoded using other code pages over to UTF-8, though non-UTF encodings are getting rarer and rarer every day

[^1]: System locale can be changed in Control panel > Region (switch view to large icons to find it) > Administrative > Change system locale.
[^2]: Beta feature toggle can also be found in the change system locale window like in footnote 1

**Comment by spytheman** [comment]
@revosw thank you for investigating in details

**Comment by spytheman** [comment]
@revosw do more recent versions of tcc on windows still use fopen or something else?
Will the issue still happen, if our tcc is compiled with gcc instead of msvc?

**Comment by revosw** [comment]
I downloaded the latest commit, `4597a96`, from https://repo.or.cz/w/tinycc.git. As I understand it, `tcc.c` is what is compiled to `tcc.exe`. And inside `tcc.c`, `int main(...)` is used
```
int main(int argc, char **argv)
{
    TCCState *s, *s1;
    // ...rest of code
}
```
And the problem here is that tcc is invoked with an input path that contains cyrillic text. Since `int main()` is used as entrypoint instead of `int wmain()`, its `argv[1]` will be narrow `char*` and not `wchar_t`, and so it relies on the ANSI code page of my system locale. We could maybe have been saved if tcc used `GetCommandLineW()` to interpret `argv[1]`, but from what I searched in the code and prompted claude, it doesn't seem that tcc itself uses it when compiling from the command line with `tcc.exe`

**Comment by medvednikov** [comment]
TCC+Windows issue.
